tuyaha/devices/climate.py

```diff
@@ -2,8 +2,52 @@
 
 
 class TuyaClimate(TuyaDevice):
```

Comment:
Looks like you're trying to fix another issue here, not related to Tuya server error problem. If yes, please, create a separate PR. Community will test them separately.

Reply:
Ok, let me better explain how this PR was created.
This are not just some changes that I test internally but the result of the test performed in last 2 months by the HA community using the custom_component [tuya custom](https://github.com/ollo69/ha_tuya_custom) that I created.
Tuya API have a lot a limitation, I try to make it better to be able to have a better control of device as light and climate, this PR is the result of test done on this component with the wish to bring back this feature in standard component
Split to another PR  will cause to me additional because need to review the code for this device leaving only some part. more complex than leave it as is.

Reply:
Then at least add a comment with explanation of what is happening here or maybe a link to github issue. This class was very simple. Now it contains a lot of `if`s, magic numbers and calculations. We need to make it clear what's happening here for future developers.

Reply:
Added comment as requested (as I should have already done). If still not clear i will add more of course....